pages/terra/srpm.mdx

```diff
@@ -0,0 +1,922 @@
+
+## Electron
+These macros assist with the general packaging of applications that use [Electron](https://www.electronjs.org).
+
+### %electronmeta
+Set up all basic requirements including build macros for building an Electron app.
+
+<Callout>
+  This does NOT install the JavaScript package managers used for builds (e.g., `nodejs-npm`, `bun`, `pnpm`, `yarnpkg`, `yarnpkg-berry`).
+  Specify these `BuildRequires:{:rpmspec}` manually.
+</Callout>
+
+Available flags:
+- `-a{:rpmspec}` will pull in additional build dependencies often needed particularly when building with NPM.\
+  Currently, this includes `nodejs-devel`, `nodejs-packaging`, and `typescript`.
+- `-D{:rpmspec}` will disable debuginfo package creation. This is necessary for many Electron apps as the bundled libraries often do not contain enough to strip, meaning this step in the build will fail. SHOULD only be used if debug packages cannot be created.
+
+Rundown of what all `%electronmeta{:rpmspec}` does:
+- Converts `%_target_cpu{:rpmspec}` (the RPM build architecture) into `%_electron_cpu{:rpmspec}`, the Electron format for compatible architectures. See the table below for conversions. See also: [%electron_arches](#electron_arches)
+  - These are defined because Electron names the build files and directories after the architecture.
+  - The [macros used to build Electron apps](#electron-build-macros) all require this to be set.
+  - In some cases, these architectures are used in the source tarballs for Electron projects.
+  - This can also be used to help work with the files if you need to do anything manually.
+- Defines `%__provides_excludes{:rpmspec}` to make sure Electron apps are not flagged as providers of their bundled dependencies, such as FFmpeg libraries.
+- Defines `%__requires_excludes{:rpmspec}` based on architecture to prevent Electron projects' cross-platform builds from causing false dependency positives from RPM.
+- If using the `-D` boolean, sets `%debug_package{:rpmspec}` to `%{nil}{:rpmspec}` to prevent stripping of bundled libraries and/or build failures due to not enough data to strip.
+- Sets the build and runtime dependencies.
+- Sets the `ExclusiveArch:{:rpmspec}` tag to only the architectures Electron supports.
+- Defines a `%bcond{:rpmspec}` for whether or not Electron is bundled for other macros.
+- Defines the bundled dependencies as bundled.
+  <Callout type="warning">
+    This only defines Electron's bundled dependencies, not any Node modules the app may also bundle.
+  </Callout>
+
+<div style={{ margin: "1em 0em 1em 0.5em" }}>
+  | `%_target_cpu`                  | `%_electron_cpu` |
+  |:-------------------------------:|:----------------:|
+  | `%x86_64`[^1]                   | `x64`            |
+  | `%arm64` (`aarch64`)            | `arm64`          |
+  | `%ix86`[^2]                     | `ia32`           |
+  | `armv7l`, `armv7hl`, `armv7hnl` | `armv7l`         |
+  | `%mips64`                       | `mips64el`       |
```

Comment:
should be noted that terra only supports x86_64 and it's microarch levels and arm64, we should pawbably make it policy that the x86 microarch level for a package shouldn't be higher than actually required